### ISSUE TITLE
$ref in properties being treated as a JSONSchema reference

### ISSUE THREAD

**Issue by DanStory** [Gi60s/openapi-enforcer#133]
We have an OpenAPI spec with definitions that describe some limited JSONSchema, as the API endpoint returns JSONSchema itself.

Updating openapi-enforcer from `1.13.0` to `1.17.0` has a bug/error when the OpenAPI spec is parsed
```
TypeError: value.$ref.split is not a function or its return value is not iterable
    at parse (.../node_modules/openapi-enforcer/src/ref-parser.js:291:49)
```
Basically the definition has a `$ref` under properties to describe an actual object property and seems to trip up thinking it is a JSONSchema reference when it isn't.
example:
```
    "definitions": {
        "Schema": {
            "properties": {
                "$ref": {
                    "type": "string"
                }
            },
            "required": [
                "$ref"
            ],
            "type": "object"
        }
    }
```

**Comment by r365-jspeirs** [comment]
Hey Dan,

Sorry for the slow response. I think I know what might be causing this and how to fix it. It should be a relatively simple fix so hopefully I'll have this published for you in the next day or two.

**Comment by Gi60s** [comment]
Hey Dan.

Sorry for the slow response.

I think I know how to fix this. Give me a few days.

**Comment by Gi60s** [comment]
I've published a fix to npm as version `1.17.1`. Thanks for reporting the issue. Let me know if there are any other problems.
